models/resnet/resnet.hpp

```diff
@@ -0,0 +1,554 @@
+          downSampleInputWidth, downSampleInputHeight));
+
+      mlpack::Log::Info << "  Convolution: " << "(" << inSize << ", " <<
+          downSampleInputWidth << ", " << downSampleInputHeight << ")" <<
+          " ---> (" << outSize << ", " << downSampleInputWidth << ", " <<
+          downSampleInputHeight << ")" << std::endl;
+    }
+    else
+    {
+      baseLayer->Add(new ann::Convolution<>(inSize, outSize, kernelWidth,
+          kernelHeight, strideWidth, strideHeight, padW, padH,
+          inputWidth, inputHeight));
+
+      mlpack::Log::Info << "Convolution: " << "(" << inSize << ", " <<
+          inputWidth << ", " << inputHeight << ")" << " ---> (";
+
+      // Updating input dimesntions.
+      inputWidth = ConvOutSize(inputWidth, kernelWidth, strideWidth, padW);
+      inputHeight = ConvOutSize(inputHeight, kernelHeight, strideHeight, padH);
+
+      mlpack::Log::Info << outSize << ", " << inputWidth << ", " << inputHeight
+          << ")" << std::endl;
+    }
+  }
+
+  /**
+   * Adds 1x1 Convolution Block and a batch norm layer to constrcut downSample
+   *     block.
+   *
+   * @tparam AddMergeType Layer type in which downSample block will
+   *     be added.
+   *
+   * @param inSize Number of input maps.
+   * @param outSize Number of output maps.
+   * @param downSampleInputWidth Input widht for down-sample block.
+   * @param downSampleInputHeight Input height for down-sample block.
+   * @param kernelWidth Width of the filter/kernel.
+   * @param kernelHeight Height of the filter/kernel.
+   * @param strideWidth Stride of filter application in the x direction.
+   * @param strideHeight Stride of filter application in the y direction.
+   * @param padW Padding width of the input.
+   * @param padH Padding height of the input.
+   */
+  template <typename AddMergeType = ann::AddMerge<>>
+  void DownSample(AddMergeType* resBlock,
+                  const size_t inSize,
+                  const size_t outSize,
+                  const size_t downSampleInputWidth,
+                  const size_t downSampleInputHeight,
+                  const size_t kernelWidth = 1,
+                  const size_t kernelHeight = 1,
+                  const size_t strideWidth = 2,
+                  const size_t strideHeight = 2,
+                  const size_t padW = 0,
+                  const size_t padH = 0)
+  {
+    ann::Sequential<>* downSampleBlock = new ann::Sequential<>();
+    ConvolutionBlock1x1(downSampleBlock, inSize, outSize, downSampleInputWidth,
+        downSampleInputHeight, strideWidth, strideHeight, kernelWidth,
+        kernelHeight, padW, padH, true);
+
+    downSampleBlock->Add(new ann::BatchNorm<>(outSize, 1e-5));
+    mlpack::Log::Info << "  BatchNorm: " << "(" << outSize << ")" << " ---> ("
+        << outSize << ")" << std::endl;
+    resBlock->Add(downSampleBlock);
+  }
+
+  /**
+   * Adds basicBlock block for ResNet 18 and 34.
+   *
+   * It's represented as:
+   * 
+   * resBlock - AddMerge layer
```

Comment:
Would be a good idea to wrap this in a `\code/endcode` that way doxygen can parse this.